db/memtable_list.cc

```diff
@@ -240,18 +240,29 @@ void MemTableListVersion::Remove(MemTable* m,
   memlist_.remove(m);
 
   m->MarkFlushed();
-  if (max_write_buffer_number_to_maintain_ > 0) {
+  if (max_write_buffer_size_to_maintain_ > 0) {
     memlist_history_.push_front(m);
     TrimHistory(to_delete);
   } else {
     UnrefMemTable(to_delete, m);
   }
 }
 
+int MemTableListVersion::ApproximateMemoryUsage() {
+  // use *parent_memtable_list_memory_usage_ instead?
+  int total_memtable_size = 0;
+  for (auto& memtable : memlist_) {
+    total_memtable_size += static_cast<int>(memtable->ApproximateMemoryUsage());
+  }
+  for (auto& memtable : memlist_history_) {
+    total_memtable_size += static_cast<int>(memtable->ApproximateMemoryUsage());
+  }
+  return total_memtable_size;
+}
+
 // Make sure we don't use up too much space in history
 void MemTableListVersion::TrimHistory(autovector<MemTable*>* to_delete) {
-  while (memlist_.size() + memlist_history_.size() >
-             static_cast<size_t>(max_write_buffer_number_to_maintain_) &&
+  while (ApproximateMemoryUsage() > max_write_buffer_size_to_maintain_ &&
```

Comment:
It might be a good idea to also have a global limit (something like db_max_write_buffer_size_to_maintain). The current approach allows users to control mem usage at the CF level, but cannot control aggregate usage.

Reply:
On second thoughts, I take this back. A global limit doesn't really help prioritize across column families. If the user doesn't want transaction history for some column families, they can tune ```max_write_buffer_size_to_maintain``` down for those.